File: output/StarCoder/codenet/C++/Python/s447383972.py
n=int(input())
a=list(map(int,input().split()))
a.sort()
tot=0
for i in range(1,n):
	if a[i]==a[i-1]:
		q[tot]=a[i]
		tot+=1
print(q[tot]*q[tot-1])

File: output/StarCoder/codenet/C++/Java/s447383972.java
import java.util.*;
public class s447383972 {
	public static void main(String[] args){
		Scanner cin=new Scanner(System.in);
		int n=cin.nextInt();
		int[] a=new int[n+1];
		for(int i=1;i<=n;i++)	a[i]=cin.nextInt();
		Arrays.sort(a);
		int tot=0;
		for(int i=1;i<=n;i++)
			if(a[i]==a[i+1])	q[++tot]=a[i],i++;
		System.out.println(q[tot]*q[tot-1]);
	}
}